modules/simulations/eureka/stack.py

```diff
@@ -0,0 +1,230 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from string import Template
+from typing import Any, cast
+
+import yaml
+from aws_cdk import Duration
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_ecr as ecr
+from aws_cdk import aws_eks as eks
+from aws_cdk import aws_sqs as sqs
+from aws_cdk import Environment, Stack, Tags
+from aws_cdk.aws_ecr_assets import DockerImageAsset
+from cdk_ecr_deployment import DockerImageName, ECRDeployment
+from cdk_nag import NagPackSuppression, NagSuppressions
+from constructs import Construct, IConstruct
+
+project_dir = os.path.dirname(os.path.abspath(__file__))
+
+ROLE_NAME = "addf-eureka-simulation-role"
+APPLICATION_IMAGE_NAME = "ubuntu-ros2"
+
+
+class EurekaStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        project_name: str,
+        deployment_name: str,
+        module_name: str,
+        stack_description: str,
+        eks_cluster_name: str,
+        eks_cluster_admin_role_arn: str,
+        eks_oidc_arn: str,
+        eks_cluster_open_id_connect_issuer: str,
+        simulation_data_bucket_name: str,
+        sqs_name: str,
+        fsx_volume_handle: str,
+        fsx_mount_point: str,
+        application_ecr_name: str,
+        env: Environment,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, id, description=stack_description, **kwargs)
+
+        self.project_name = project_name
+        self.deployment_name = deployment_name
+        self.module_name = module_name
+
+        dep_mod = f"{self.project_name}-{self.deployment_name}-{self.module_name}"
+        dep_mod = dep_mod[:64]
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=dep_mod)
+
+        provider = eks.OpenIdConnectProvider.from_open_id_connect_provider_arn(
+            self, f"{dep_mod}-provider", eks_oidc_arn
+        )
+        eks_cluster = eks.Cluster.from_cluster_attributes(
+            self,
+            f"{dep_mod}-eks-cluster",
+            cluster_name=eks_cluster_name,
+            kubectl_role_arn=eks_cluster_admin_role_arn,
+            open_id_connect_provider=provider,
```

Comment:
Can you import the Kubectl29Layer and mention it here, so that the default layer is not used? If the default layer is used, we are seeing some unexpected errors related to restmappings.

example is found here: https://github.com/awslabs/autonomous-driving-data-framework/blob/main/modules/integration/efs-on-eks/stack_efs_eks.py#L57

Reply:
Addressed